lib/active_merchant/billing/gateways/datatrans.rb

```diff
@@ -196,26 +231,36 @@ def headers
       def url(endpoint, options = {})
         case endpoint
         when 'settle', 'credit', 'cancel'
-          "#{test? ? test_url : live_url}#{options[:transaction_id]}/#{endpoint}"
+          "#{test? ? test_url : live_url}transactions/#{options[:transaction_id]}/#{endpoint}"
+        when 'tokenize'
+          "#{test? ? test_url : live_url}aliases/#{endpoint}"
+        when 'delete_alias'
+          "#{test? ? test_url : live_url}aliases/#{options[:alias_id]}"
         else
-          "#{test? ? test_url : live_url}#{endpoint}"
+          "#{test? ? test_url : live_url}transactions/#{endpoint}"
         end
       end
 
       def success_from(action, response)
         case action
         when 'authorize', 'credit'
-          true if response.include?('transactionId') && response.include?('acquirerAuthorizationCode')
+          response.include?('transactionId') && response.include?('acquirerAuthorizationCode')
         when 'settle', 'cancel'
-          true if response.dig('response_code') == 204
+          response.dig('response_code') == 204
+        when 'tokenize'
+          response['responses'][0].include?('alias') && response['overview']['failed'] == 0
+        when 'delete_alias'
+          response.dig('response_code') == 204
         else
           false
         end
       end
 
-      def authorization_from(response)
-        auth = [response['transactionId'], response['acquirerAuthorizationCode']].join('|')
-        return auth unless auth == '|'
+      def authorization_from(response, action, options)
+        token_array = [response.dig('responses', 0, 'alias'), options[:expiry_month], options[:expiry_year]] if action == 'tokenize'
```

Comment:
💬 👃 Comment/Code smell:

Event when implementation seems correct, you can see that token_array should be just one value maybe something like "<token_xxxxxx>-01-2025" so when you see the "xxxxxx|xxxxx|<token_xxxxxx>-01-2025" you can know that those 3 values are independent from each other, an no interpretation is needed.

Reply:
I was dubbing about it hehe